storage/innobase/xtrabackup/src/redo_log.cc

```diff
@@ -776,6 +785,51 @@ void Archived_Redo_Log_Monitor::thread_func() {
       hdr_len -= n_read;
     }
 
+    /* for 8.0.22 onwards first block of redo archive file is all zeros */
+    if (log_block_get_checksum(buf) == 0) {
+      archive_first_block_zero = true;
+      hdr_len = QUEUE_BLOCK_SIZE - OS_FILE_LOG_BLOCK_SIZE;
+      aligned_array_pointer<byte, UNIV_PAGE_SIZE_MAX> buf2;
+      buf2.create(QUEUE_BLOCK_SIZE - OS_FILE_LOG_BLOCK_SIZE);
+      while (hdr_len > 0 && !stopped) {
+        size_t n_read = my_read(file, buf2, hdr_len, MYF(MY_WME));
+        if (n_read == MY_FILE_ERROR) {
+          msg("xtrabackup: cannot read from '%s'\n", archive.filename.c_str());
+          mysql_close(mysql);
+          my_thread_end();
+          return;
+        }
+        os_event_wait_time_low(event, 100 * 1000, 0);
+        os_event_reset(event);
+        hdr_len -= n_read;
+      }
+
+      hdr_len = OS_FILE_LOG_BLOCK_SIZE;
+      while (hdr_len > 0 && !stopped) {
+        size_t n_read = my_read(file, buf, hdr_len, MYF(MY_WME));
+        if (n_read == MY_FILE_ERROR) {
+          msg("xtrabackup: cannot read from '%s'\n", archive.filename.c_str());
+          mysql_close(mysql);
+          my_thread_end();
+          return;
+        }
+        os_event_wait_time_low(event, 100 * 1000, 0);
+        os_event_reset(event);
+        hdr_len -= n_read;
+      }
+      if (srv_redo_log_encrypt) {
+        aligned_array_pointer<byte, OS_FILE_LOG_BLOCK_SIZE> scratch_buf;
+        scratch_buf.create(OS_FILE_LOG_BLOCK_SIZE);
+        IORequest req_type(IORequestLogRead);
+        fil_space_t *space = fil_space_get(dict_sys_t::s_log_space_first_id);
+        fil_io_set_encryption(req_type, page_id_t(space->id, 0), space);
+        Encryption encryption(req_type.encryption_algorithm());
+        auto err = encryption.decrypt_log(req_type, buf, hdr_len, scratch_buf,
+                                          UNIV_PAGE_SIZE_MAX);
```

Comment:
the scratch_buf is only OS_FILE_LOG_BLOCK_SIZE. UNIV_PAGE_SIZE_MAX is wrong here